src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -143,26 +239,33 @@ public void UpdateWorkloads(bool includePreviews = false, DirectoryPath? offline
             transaction.Run(
                 action: context =>
                 {
+                    UpdateInstalledWorkloadsFromHistory(sdkFeatureBand, context, offlineCache);
```

Comment:
If we do this before updating the manifests, then it will potentially install the old versions of a workload and then have to install the new version after updating the manifests.  I think this should be after the manifest update.

Reply:
It occurred to me when you said that that it probably should also only execute if !_historyManifestOnlyOption...